pandas/core/arrays/base.py

```diff
@@ -551,7 +558,7 @@ def astype(self, dtype, copy=True):
         ):  # allow conversion to StringArrays
             return dtype.construct_array_type()._from_sequence(self, copy=False)
 
-        return np.array(self, dtype=dtype, copy=copy)
+        return np.array(self, dtype=cast(NpDtype, dtype), copy=copy)
```

Comment:
can we guarantee that dtype is a numpy dtype at this point? I think better remove the cast to leave any error messages about this for visibility.

we do raise a TypeError so mypy is correct here and this is not a false positive. we don't want to lie to mypy here to suppress the warning.

```
>>> s = pd.Series(pd.date_range("2020", periods=3, tz="Europe/London"))
>>> s
0   2020-01-01 00:00:00+00:00
1   2020-01-02 00:00:00+00:00
2   2020-01-03 00:00:00+00:00
dtype: datetime64[ns, Europe/London]
>>> 
>>> dta = s._values
>>> dta
<DatetimeArray>
['2020-01-01 00:00:00+00:00', '2020-01-02 00:00:00+00:00',
 '2020-01-03 00:00:00+00:00']
Length: 3, dtype: datetime64[ns, Europe/London]
>>> 
>>> dta.astype("boolean")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/simon/pandas/pandas/core/arrays/datetimes.py", line 626, in astype
    return dtl.DatetimeLikeArrayMixin.astype(self, dtype, copy)
  File "/home/simon/pandas/pandas/core/arrays/datetimelike.py", line 458, in astype
    return np.asarray(self, dtype=dtype)
  File "/home/simon/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/numpy/core/_asarray.py", line 102, in asarray
    return array(a, dtype, copy=False, order=order)
TypeError: Cannot interpret 'BooleanDtype' as a data type
>>> 
```

Reply:
I see your point here.  Will change it and keep the mypy error.